packages/nextjs/test/config.test.ts

```diff
@@ -0,0 +1,220 @@
+ * that.
+ *
+ * @param options An object including the following:
+ *   - `userNextConfig` Next config options provided by the user
+ *   - `userSentryWebpackPluginConfig` SentryWebpackPlugin options provided by the user
+ *   - `incomingWebpackConfig` The existing webpack config, passed to the function as `config`
+ *   - `incomingWebpackBuildContext` The existing webpack build context, passed to the function as `options`
+ *
+ * @returns The webpack config values next will use when it calls the function that `createFinalWebpackConfig` returns
+ */
+async function materializeFinalWebpackConfig(options: {
+  userNextConfig: ExportedNextConfig;
+  userSentryWebpackPluginConfig: SentryWebpackPluginOptions;
+  incomingWebpackConfig: WebpackConfigObject;
+  incomingWebpackBuildContext: BuildContext;
+}): Promise<WebpackConfigObject> {
+  const { userNextConfig, userSentryWebpackPluginConfig, incomingWebpackConfig, incomingWebpackBuildContext } = options;
+
+  // get the webpack config function we'd normally pass back to next
+  const webpackConfigFunction = constructWebpackConfigFunction(userNextConfig, userSentryWebpackPluginConfig);
+
+  // call it to get concrete values for comparison
+  const finalWebpackConfigValue = webpackConfigFunction(incomingWebpackConfig, incomingWebpackBuildContext);
+  const webpackEntryProperty = finalWebpackConfigValue.entry as EntryPropertyFunction;
+  finalWebpackConfigValue.entry = await webpackEntryProperty();
+
+  return finalWebpackConfigValue;
+}
+
+describe('withSentryConfig', () => {
+  it('includes expected properties', () => {
+    const finalConfig = materializeFinalNextConfig(userNextConfig, userSentryWebpackPluginConfig);
+
+    expect(finalConfig).toEqual(
+      expect.objectContaining({
+        productionBrowserSourceMaps: true,
+        webpack: expect.any(Function), // `webpack` is tested specifically elsewhere
+      }),
+    );
+  });
+
+  it('preserves unrelated next config options', () => {
+    const finalConfig = materializeFinalNextConfig(userNextConfig, userSentryWebpackPluginConfig);
+
+    expect(finalConfig.publicRuntimeConfig).toEqual(userNextConfig.publicRuntimeConfig);
+  });
+
+  it("works when user's overall config is an object", () => {
+    const finalConfig = materializeFinalNextConfig(userNextConfig, userSentryWebpackPluginConfig);
+
+    expect(finalConfig).toEqual(
+      expect.objectContaining({
+        ...userNextConfig,
+        productionBrowserSourceMaps: true,
+        webpack: expect.any(Function), // `webpack` is tested specifically elsewhere
+      }),
+    );
+  });
+});
+
+describe('webpack config', () => {
+  beforeEach(() => {
+    // nuke this so it won't try to look for our dummy paths
+    (storeServerConfigFileLocation as jest.Mock).mockImplementationOnce(() => undefined);
+  });
+
+  it('includes expected properties', async () => {
+    const finalWebpackConfig = await materializeFinalWebpackConfig({
+      userNextConfig,
```

Comment:
I feel like we should table test both the `includes expected properties` and `injects correct code when building client bundle` tests. We can create various different server/client configs and buildContexts and check to make sure we get a sane finalWebpackConfig.

See my table tests for sessions as an example: https://github.com/getsentry/sentry-javascript/blob/master/packages/hub/test/session.test.ts#L32-L90.

We can also come back to this once we update the `require` server init flow, feel free to slap a TODO somewhere.

Reply:
Yeah, a table test isn't a bad idea. We can cross that bridge once we have enough different scenarios to warrant it.